fortuna/src/config.rs

```diff
@@ -105,7 +110,7 @@ impl Config {
         // Open and read the YAML file
         // TODO: the default serde deserialization doesn't enforce unique keys
         let yaml_content = fs::read_to_string(path)?;
-        let config: Config = serde_yaml::from_str(&yaml_content)?;
+        let config: Config = serde_yaml::from_str(&Config::inject_env(&yaml_content)?)?;
```

Comment:
I think you should explicitly pass a map of substitutions into the `load` function and down into the config replacement. That will make these functions into pure functions (which are easy to test, use in other contexts, etc)

Reply:
this also forces you to make the dependencies of the code explicit instead of implicit